### PR TITLE
[-] Models - Improve models exports to support model addition / deletion without requiring changes.

### PR DESCRIPTION
**Pull request by SteveBunlon**
For this PR I would suggest to not change the changelog, and to squash the present work with this one : https://github.com/ForestAdmin/lumber/pull/386

As the subjects are related and next to each other in the git history, It would be much cleaner and tinier.

### CODE REVIEW

**Comment by SteveBunlon** [comment]
[P3 - Lumber - The current way of exporting / importing models is broken (mongoose)](https://trello.com/c/hrG67LrN/131-p3-lumber-the-current-way-of-exporting-importing-models-is-broken-mongoose)

File: templates/app/models/index.hbs [hunk 1]
```diff
@@ -1,16 +1,21 @@
...
 
-const models = {
-{{#each models as |model|}}
-  {{model.modelName}},
-{{/each}}
-};
+mongoose.connect(process.env.DATABASE_URL, { useNewUrlParser: true });
+
+const modelsList = requireAll({
+  dirname:  __dirname,
```
Comment (arnaudbesnier) [inline templates/app/models/index.hbs]:
```suggestion
  dirname: __dirname,
```
?

File: templates/app/models/index.hbs [hunk 2]
```diff
@@ -1,16 +1,21 @@
...
+  filter: `^(?!index\.js$).+`,
+  resolve: (Model) => Model,
```
Comment (arnaudbesnier) [inline templates/app/models/index.hbs]: Maybe `recursive: true` would be a good addition for more flexible code in the future.

File: templates/app/models/index.hbs [hunk 3]
```diff
@@ -1,16 +1,21 @@
...
-const models = {
-{{#each models as |model|}}
-  {{model.modelName}},
-{{/each}}
-};
+mongoose.connect(process.env.DATABASE_URL, { useNewUrlParser: true });
+
+const modelsList = requireAll({
+  dirname:  __dirname,
+  filter: `^(?!index\.js$).+`,
```
Comment (arnaudbesnier) [inline templates/app/models/index.hbs]: Following what we have in `forest-express` `requireAll`, maybe:
```javascript
  excludeDirs: /^__tests__$/,
  filter: (fileName) => isJavascriptFileName(fileName) && !isTestFileName(fileName),
```
could be a good improvement.

File: templates/app/models/index.hbs [hunk 4]
```diff
@@ -1,16 +1,21 @@
 {{#if isMongoDB}}
 const mongoose = require('mongoose');
-{{#each models as |model|}}
-const {{model.modelName}} = require('./{{model.modelFile}}');
-{{/each}}
+const requireAll = require('require-all');
 
-mongoose.connect(process.env.DATABASE_URL, { useNewUrlParser: true, useUnifiedTopology: true });
```
[the remark this answers is not in the document]
Comment (SteveBunlon) [reply templates/app/models/index.hbs]: Wrong copy past i guess, this is not needed, I added it back.

File: templates/app/models/index.hbs (hunk 2 above)
Comment (SteveBunlon) [reply templates/app/models/index.hbs]: As discussed together, this will be handled in the future.

File: templates/app/models/index.hbs (hunk 3 above)
Comment (SteveBunlon) [reply templates/app/models/index.hbs]: As discussed together, this will be handled in the future.

**Pull request merged**